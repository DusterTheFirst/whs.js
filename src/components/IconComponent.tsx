/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { Platform, View } from "react-native";
import IonIcon from "react-native-vector-icons/Ionicons";

export default function IconComponent({ name, size = 22 }: { name: string; size?: number }) {
    return (
        <View style={{marginBottom: -4}}>
            <IonIcon
                name={`${Platform.OS === "ios" ? "ios" : "md"}-${name}`}
                size={size}
                color={"#2f95dc"}
            />
        </View>
    );
}