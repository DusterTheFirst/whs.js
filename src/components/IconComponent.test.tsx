/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { render } from "react-native-testing-library";
import IconComponent from "./IconComponent";

describe("Tests different variations of <IconComponent/>", () => {
    it("Renders <IconComponent name=\"list\"/>", () => {
        let comp = render(<IconComponent name="list"/>);

        expect(comp.toJSON()).toMatchSnapshot();
    });
    it("Renders <IconComponent name=\"cog\" size={20}/>", () => {
        let comp = render(<IconComponent name="cog" size={20}/>);

        expect(comp.toJSON()).toMatchSnapshot();
    });
});