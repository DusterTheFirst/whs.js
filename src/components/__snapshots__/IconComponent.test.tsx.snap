// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests different variations of <IconComponent/> Renders <IconComponent name="cog" size={20}/> 1`] = `
<View
  style={
    Object {
      "marginBottom": -4,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#2f95dc",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`Tests different variations of <IconComponent/> Renders <IconComponent name="list"/> 1`] = `
<View
  style={
    Object {
      "marginBottom": -4,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#2f95dc",
          "fontSize": 22,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
